cache/contenthash/checksum_test.go

```diff
@@ -107,15 +108,15 @@ func TestChecksumHardlinks(t *testing.T) {
 	err = emit(cc2.HandleChange, changeStream(ch))
 	require.NoError(t, err)
 
-	dgst, err = cc2.Checksum(context.TODO(), ref, "abc/foo", false)
+	dgst, err = cc2.Checksum(context.TODO(), newBadMountable(), "abc/foo", false)
```

Comment:
It's confusing why these old test cases (that don't even return error) use `newBadMountable()`. I guess in these cases `Mount` does not get called but this is a side-effect that shouldn't be tested. I think just leave it to the new test. If you want to add more error cases add new conditions and/or comments why `newBadMountable` is passed.

Reply:
Should it not be tested? What I was trying to do here, beyond justifying `newBadMountable`, was have the tests ensure that if given a `changeStream`, it doesn't then re-hit the disk needlessly. That seems like the point of the cache, rather than a side-effect.

Reply:
If the noise in the tests is a concern, I could move this logic into `createRef`, since it's the inverse of the Windows skip-test: Windows _currently_ skips tests that intend to touch the disk, and I think it's worth having the test ensure that tests that should not touch the disk, do not touch the disk.

Reply:
Hmm, That would make `badMountable` more complex. I'd have to promote it to a `cache.ImmutableRef` wrapper, since we call other functions than `Mount` on it. That seems like it wouldn't be hard to do though. Just noisier, and the failure-reason (if it ever fails) is no longer written on the line that generated the error.

Reply:
If we want to test the lru cache behavior there should be a special test for that. Overall, LRU is for performance so it could change in the future so that mount gets called (especially because `Mount` itself is not expensive) but overall checksumming still remains performant.

Reply:
Ah, I didn't realise this was an LRU cache. I've reverted the changes to the existing tests.